### PR TITLE
add translation module 

### PR DESCRIPTION
**Pull request by eastonsuo**
not sure the if the code is correct, is there some way to test it. for example to generate .pot file locally?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -118,13 +119,16 @@ const extension: JupyterFrontEndPlugin<void> = {
...
+  requires: [IFileBrowserFactory, ISettingRegistry, ITranslator],
 
   activate: (
     app: JupyterFrontEnd,
     factory: IFileBrowserFactory,
-    settingRegistry: ISettingRegistry
+    settingRegistry: ISettingRegistry,
+    translator: ITranslator
   ) => {
+    const _trans = translator.load('jupyter_archive');
```
Comment (krassowski) [inline src/index.ts:130]: Needs to be named `trans` without underscore.

**Comment by krassowski** [comment]
It looks good! Just needs to rename the constant here.

File: src/index.ts [hunk 2]
```diff
@@ -277,7 +281,7 @@ const extension: JupyterFrontEndPlugin<void> = {
       label: args => {
         const format = (args['format'] as ArchiveFormat) || '';
         const label = format.replace('.', ' ').toLocaleUpperCase();
-        return label ? `${label} Archive` : 'Download as an Archive';
+        return label ? `${label} Archive` : _trans.__('Download as an Archive');
```
Comment (fcollonval) [inline src/index.ts:284]: You can set a parametrize string as translatable using the following syntax:

```suggestion
        return label ? trans.__('%1 Archive', label) : trans.__('Download as an Archive');
```

File: src/index.ts [hunk 3]
```diff
@@ -310,7 +314,7 @@ const extension: JupyterFrontEndPlugin<void> = {
         }
         return visible;
       },
-      label: 'Extract Archive'
+      label: _trans.__('Extract Archive')
```
Comment (fcollonval) [inline src/index.ts:317]:
```suggestion
      label: trans.__('Extract Archive')
```

File: src/index.ts [hunk 4]
```diff
@@ -341,7 +345,7 @@ const extension: JupyterFrontEndPlugin<void> = {
         const label = format.replace('.', ' ').toLocaleUpperCase();
         return label
           ? `${label} Archive`
-          : 'Download Current Folder as an Archive';
+          : _trans.__('Download Current Folder as an Archive');
```
Comment (fcollonval) [inline src/index.ts:348]: This needs to be changed as earlier:

```ts
? trans.__('%1 Archive', label)
: trans.__('Download Current Folder as an Archive');
```

File: package.json [hunk 5]
```diff
@@ -65,6 +65,7 @@
   },
   "devDependencies": {
     "@jupyterlab/builder": "^3.0.0",
+    "@jupyterlab/translation": "^3.0.0",
```
Comment (fcollonval) [inline package.json:68]: This is a `dependencies` not a development dependency as the translation needs to be applied at runtime.

File: yarn.lock [hunk 6]
```diff
@@ -263,6 +263,19 @@
     path-browserify "^1.0.0"
     url-parse "~1.5.1"
 
+"@jupyterlab/coreutils@^5.1.13":
```
Comment (fcollonval) [inline yarn.lock:266]: The `yarn.lock` should not need changes. Would you mind restoring the `master` version and check it does not get updated with `@jupyterlab/translation` added as dependency?

**Review by fcollonval — CHANGES_REQUESTED** [review]
Thanks @eastonsuo to quickly open a PR

I left some comments; mainly related to the need to use `trans` as variable name as `_trans` won't be recognized.

**Review by fcollonval — APPROVED** [review]

**Comment by fcollonval** [comment]
Thanks a lot @eastonsuo for the quick corrections.

**Pull request merged**